eth_account/account.py

```diff
@@ -1043,3 +1054,99 @@ def sign_typed_data(
+        It adds the signature fields to the authorization dict.
+
+        :param dict authorization_dict: the required keys are: chainId, address, nonce
+        :param private_key: the private key to sign the data with
+        :type private_key: hex str, bytes, int or :class:`eth_keys.datatypes.PrivateKey`
+        :returns: the dictionary with the signature fields added, suitable for inclusion in a EIP-7702 transaction
+        :rtype: dict
+
+        .. NOTE::
+            You need to get signed one or more signed authorizations from an EOA willing to have a smart contract code associated with the EOA, this the essence of EIP-7702
+
+            in the code below in the variable authorization_to_sign:
+            - chainId is the chain id of the chain where the EOA is located or 0 if the authorization is for all chains
+            - address is the address of the smart contract code to be associated with the EOA, the address format is bytes
+            - nonce is the nonce of the EOA, it is used to prevent replay attacks
+            once signed, the fields are the signature of the first 3 fields by the EOA
+
+            to create a transaction that associates the code with the EOA, you need to create a transaction with the authorizationList field, this field is a list of signed authorizations, one for each EOA willing to have the code associated with its address
+
+        .. doctest:: python
+
+            >>> from eth_account import Account
+
+            >>> key = "0xaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"
+            >>> auth = {
+            ...     "chainId": 1337,
+            ...     "address": "0x5ce9454909639d2d17a3f753ce7d93fa0b9ab12e",
+            ...     "nonce": 0,
+            ... }
+
+            >>> auth_signed = Account.sign_authorization(auth, key)
+            >>> auth_signed
+            SignedSetCodeAuthorization(chain_id=1337,
+             address=b'\\\xe9EI\tc\x9d-\x17\xa3\xf7S\xce}\x93\xfa\x0b\x9a\xb1.',
+             nonce=0,
+             y_parity=0,
+             r=13559089679153465389530080958732165835506259206033627652019224547096284912193,
+             s=4909893498344689684383026337158184903862784607939660326835239278207516636646,
+             signature='0x1dfa2c3c5b3bf61e364f7ce7ed744aef498fe26c981de726b139e0461c545a410adae6b7a1ff6137fb698a9973f4706cf8fe56aba87e8241b76042af587175e600',
+             authorization_hash=HexBytes('0x5a1bfd35dee8528cbcbe2dad796fec10b7ad15fa67ecc77c427d0d57bc5e2737'))
+
+            >>> tx = {
+            ...     "gas": 100000,
+            ...     "maxFeePerGas": 2000000000,
+            ...     "maxPriorityFeePerGas": 2000000000,
+            ...     "data": "0x616263646566",
+            ...     "nonce": 34,
+            ...     "to": "0x09616C3d61b3331fc4109a9E41a8BDB7d9776609",
+            ...     "value": "0x5af3107a4000",
+            ...     "accessList": (),
+            ...     "authorizationList": [auth_signed.as_rpc_object()],
```

Comment:
This feels clunky. Would it be a huge pain to handle this on the other side so the user can just pass `auth_signed`?

Reply:
I currently have [an issue open](https://github.com/ethereum/web3.py/issues/3650) to be able to deprecate this and handle it implicitly in `web3.py`. I'm not really sure how else it can be "not clunky". Open to ideas.

Reply:
Come to think of it, I think this probably needs to be addressed in the `from_dict()` of the `SetCodeTransaction`. I'm going to see how to best structure this but it may very well be that to sign the transaction in the first place, it always passes through `from_dict`.

Reply:
Clunky it was indeed. Turns out my last comment was correct. We never really need to get all the way over to `web3.py` for this conversion, it should be happening inside `from_dict()`. I updated it here and added tests to make sure that the Pydantic classes are properly serialized as JSON-RPC objects.